wasm.md

```diff
@@ -1080,24 +1073,24 @@ Currently, only one memory may be accessible to a module, and thus the `<mAddr>`
 
 ```k
     syntax Defn       ::= MemoryDefn
-    syntax MemoryDefn ::= "(" "memory"                            ")"
-                        | "(" "memory"     Int                    ")" // Size only
-                        | "(" "memory"     Int Int                ")" // Min and max.
-                        | "(" "memory" "(" "data" DataStrings ")" ")"
-                        |     "memory" "{" Int MaxBound "}"
- // -------------------------------------------------------
-    rule <k> ( memory                 ) => memory { 0   .MaxBound } ... </k>
-    rule <k> ( memory MIN:Int         ) => memory { MIN .MaxBound } ... </k>
+    syntax MemoryDefn ::= "(" "memory" OptionalId Limits                     ")"
+                        | "(" "memory" OptionalId "(" "data" DataStrings ")" ")"
+                        |     "memory" "{" OptionalId Int MaxBound "}"
+ // ------------------------------------------------------------------
+    rule <k> ( memory ID:OptionalId MIN:Int         ) => memory { ID MIN .MaxBound } ... </k>
       requires MIN <=Int #maxMemorySize()
-    rule <k> ( memory MIN:Int MAX:Int ) => memory { MIN MAX       } ... </k>
+    rule <k> ( memory ID:OptionalId MIN:Int MAX:Int ) => memory { ID MIN MAX       } ... </k>
       requires MIN <=Int #maxMemorySize()
        andBool MAX <=Int #maxMemorySize()
-    rule <k> ( memory ( data DS ) )
-          =>  memory { #lengthDataPages(DS) #lengthDataPages(DS) }
-          ~> ( data (i32.const 0) DS ) ... </k>
+    rule <k> ( memory ( data DS ) ) => ( memory #freshId(NEXTID) (data DS) ) ... </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
```

Comment:
It is a little inconsistent for the way you handled `func` with no ID and `memory` with no ID.
As here you used `( memory ( data DS ) )` and when defining function you used `rule <k> ( func ID:OptionalId FEXPO:FuncExports TUSE:TypeUse LDECLS:LocalDecls INSTRS:Instrs )... requires notBool isIdentifier(ID)`. But I think they could be written in the same way as they are pretty similar cases.

Reply:
Good point, done.